torch/cuda/__init__.py

```diff
@@ -1693,6 +1694,73 @@ def addmm_kernel_impl(*args, **kwargs):
 _lazy_call(_register_triton_kernels)
 
 
+def _compile_kernel(
+    kernel_source: str,
+    kernel_name: str,
```

Comment:
This imposes wrong assumption that `kernel_source` must have only one kernel, but it could multiple. But I guess this could be addressed in followup PRs

Reply:
Indeed this is a significant limitation right now, will add this to PR description with future work